### DISCUSSION TITLE
Benchmark. Why is ImageMagick so slow ?

### DISCUSSION THREAD

**Discussion by jptheking** [ImageMagick/ImageMagick#2625] [Help]
Hello,

I recently downloaded ImageMagick-7.0.10-29-Q16-x86-static.exe.

My goal was to search for a subimage inside a larger one.
- large image: 1920x1080
- subimage: 80x80

As expected and mentionned in your help files, the process is extremely slow.

On the other side, I currently use Macro Scheduler for this task as well (30 day free version available).

I ran the tests on two different windows systems. Here are the results:

PC#1:
ImageMagick:	63 secs
Macroscheduler:	0.3 secs

PC#2
ImageMagick:	404 secs
Macroscheduler:	1.6 secs

That makes over a 200x ratio beetween the two softwares.

I suppose ImageMagick was written in low-level langage (c++?).
Can you explain this huge difference??

Thanks for your help

**Comment by snibgo** [comment]
I don't know how Macroscheduler searches. Perhaps it uses multi-scale searches, like my "-process srchimg" or script srchImg.bat. These are massively faster that IM's "-subimage-search".

See [Searching an image](http://im.snibgo.com/srchimg.htm) and [Searching an image aggressively](http://im.snibgo.com/srchimgag.htm) and the processes that search in [Process modules](http://im.snibgo.com/customim.htm).

**Comment by fmw42** [comment]
Perhaps they use a totally different approach. For example, I have bash unix normalized cross correlation script that process orders of magnitude faster using FFT processing. See http://www.fmwconcepts.com/imagemagick/linkcounter.php?dir=./&linkname=ACCELERATED_TEMPLATE_MATCHING.pdf

**Comment by jptheking** [comment]
Thank you for your answers. This seems promising.
Still the question is:
Is this is easily achievable, why not include this function in ImageMagick instead of the present (slow) one?
I am not a programmer, just a user... And a simple command line would be appreciated.
Sorry for that and thanks again.

**Reply by fmw42** [reply]
The IM developers have limited time and concentrate on bug fixes over enhancements. It is on the list to implement at a later time. If you are on a unix-like system, you could try my script. If you need help, let me know.
